sdk/src/Core/Amazon.Runtime/Internal/Util/StringUtils.cs

```diff
@@ -91,13 +91,27 @@ public static string FromDateTimeToISO8601(DateTime value)
         {
             return value.ToUniversalTime().ToString(AWSSDKUtils.ISO8601DateFormat, CultureInfo.InvariantCulture);
         }
+        
         /// <summary>
-        /// Converts a DateTime to ISO8601 formatted string.
+        /// Converts a DateTime to ISO8601 formatted string without milliseconds.
         /// </summary>
         public static string FromDateTimeToISO8601NoMs(DateTime value)
         {
             return value.ToUniversalTime().ToString(AWSSDKUtils.ISO8601DateFormatNoMS, CultureInfo.InvariantCulture);
         }
+
+        /// <summary>
+        /// Converts a DateTime to ISO8601 formatted string with milliseconds
+        /// if they are not zero.
+        /// </summary>
+        public static string FromDateTimeToISO8601Minimal(DateTime value)
```

Comment:
It will use minimal length. Meaning if there are ms != `000` it will include it else it won't. I am open to other names but the above one isn't completely correct as it may not have ms. We also already have a with and without ms method but not a smart one that includes ms if not 0.

Reply:
Yeah that is hard to name. Maybe `FromDateTimeToISO8601WithOptionalMs`? Either way it is going to come down to the docs which you do have. I'll approve the PR.

Reply:
Updated to the new suggested method name `FromDateTimeToISO8601WithOptionalMs` as it does make it clearer and is accurate.